packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2054,7 +2054,11 @@ abstract class BuildContext {
 
   /// Determines if [widget] is in the process of updating.
   ///
-  /// This flag will return `true` if [widget] is in one of those life-cycles:
+  /// This flag will return `true` if [widget] is in a life-cycles that both:
+  /// - declaratively updates a tree ([Widget] tree, [RenderObject] tree)
```

Comment:
This doesn't really help clear things up for me.  I'm not sure this matches vocabulary we use elsewhere - in fact grepping the repo for `declarative` returns 0 results today.  What's special about this?

Reply:
Perhaps looking at the docs on `performRebuild` or `rebuild`  would provide some inspiration here?